warp10/src/main/java/io/warp10/script/WarpScriptSingleValueFillerFunction.java

```diff
@@ -23,10 +23,10 @@ public interface WarpScriptSingleValueFillerFunction {
   public interface Precomputable extends WarpScriptSingleValueFillerFunction {
     public WarpScriptSingleValueFillerFunction compute(GeoTimeSerie gts) throws WarpScriptException;
 
-    default public Object evaluate(long tick) throws WarpScriptException {
+    default public void fillTick(long tick, GeoTimeSerie gts, Object invalidValue) throws WarpScriptException {
       throw new WarpScriptException("Invalid Filler Error: evaluator function has not been precomputed yet.");
     }
   }
 
-  public Object evaluate(long tick) throws WarpScriptException;
+  public void fillTick(long tick, GeoTimeSerie gts, Object invalidValue) throws WarpScriptException;
```

Comment:
This is a breaking change since you are radically changing the interface. Instead mark evaluate as deprecated, add default implementations to evaluate and fillTick which throw an exception and you should be good to go except for backward compatibility across 3.x of extensions making use of WarpScriptSingleValueFillerFunction as defined in this PR.

Reply:
I believe fillTick should default to evaluate instead of throwing an error so existing extensions are not broken.

Reply:
Existing extensions, if any, implements evaluate(), so the default method will never be executed, will it ?

Reply:
Existing extensions implements evaluate(), but your changes to fill() method in GTSHelper makes them use their default fillTick(), which then should default to their implementation of evaluate in order not to break their actual behaviour